errbot/backends/irc.py

```diff
@@ -408,7 +420,9 @@ def send_public_message(self, to, line):
         except ServerNotConnectedError:
             pass  # the message will be lost
 
-    def on_disconnect(self, _, e):
+    def on_disconnect(self, connection, event):
+        del self.bot._rooms  # Destroy all rooms on Disconnect
+        self.bot._rooms = {}
```

Comment:
The `del self.bot_rooms` line (424) may be removed entirely, as you're setting it to an empty dict on the next line (425).